src/fluree/db/nameservice/filesystem.cljc

```diff
@@ -1,85 +1,225 @@
 (ns fluree.db.nameservice.filesystem
   (:require [fluree.db.nameservice.proto :as ns-proto]
+            [fluree.db.util.core :as util :refer [try* catch*]]
             [fluree.db.util.filesystem :as fs]
             [fluree.db.util.bytes :as bytes]
             [clojure.core.async :as async :refer [go]]
             [fluree.db.util.async :refer [<? go-try]]
             [clojure.string :as str]
-            [fluree.db.util.log :as log]))
+            [fluree.db.util.json :as json]
+            [fluree.db.util.log :as log]
+            #?(:clj [clojure.java.io :as io])))
 
 #?(:clj (set! *warn-on-reflection* true))
 
+(defn ns-record
+  "Generates nameservice metadata map for JSON storage..
+
+  For now, since we only have a single branch possible,
+  always sets default-branch. Eventually will need to merge
+  changes from different branches into existing metadata map"
+  [ns-address commit-address
+   {alias  "alias"
+    branch "branch"
+    :as    json-ld-commit}]
+  (let [branch-iri (str ns-address "(" branch ")")]
+    {"@context"      "https://ns.flur.ee/ledger/v1"
+     "@id"           ns-address
+     "defaultBranch" branch-iri
+     "ledgerAlias"   alias
+     "branches"      [{"@id"     branch-iri
+                       "address" commit-address
+                       "commit"  json-ld-commit}]}))
+
 (defn address-path
   [address]
   (let [[_ _ path] (str/split address #":")]
     path))
 
-(defn address-full-path
-  [local-path address]
-  (str local-path "/" (address-path address)))
-
-(defn file-address
-  "Turn a path or a protocol-relative URL into a fluree file address."
+(defn list-files-in-path
```

Comment:
This seems like it could live alongside the other file system utils in `fluree.db.util.filesystem`.